blueflood-core/src/main/java/com/rackspacecloud/blueflood/cache/LocatorCache.java

```diff
@@ -103,6 +103,17 @@ public synchronized boolean isLocatorCurrentInDiscoveryLayer(Locator loc) {
     }
 
     /**
+     * Checks if Locator is recently inserted in the token discovery layer
+     *
+     * @param loc
+     * @return
+     */
+    public synchronized boolean isLocatorCurrentInTokenDiscoveryLayer(Locator loc) {
```

Comment:
Since writing to tokens is a separate thread pool, there is a chance, a metric is written to `metric_metadata` but not to `metric_tokens`. So tracking both of them separately.